### PR TITLE
Add SBOM Test to GitHub Workflow

### LINKED ISSUE #29443: SBOM generation to go along w/ every release
### Parent Issue

_No response_

### Task

We need to automate the generation of Software Bill of Materials (SBOM) using GitHub Actions. This automation should ensure that a new SBOM is generated with each new build and release. The preferred SBOM formats are SPDX and CycloneDX.

Additional Information:
- [SPDX Specification](https://spdx.dev/specifications/)
- [CycloneDX Specification](https://cyclonedx.org/specification/overview/)

### Proposed Objective

Security & Privacy

### Proposed Priority

Priority 2 - Important

### Acceptance Criteria

1. A GitHub Actions workflow is created and triggers on every push to the main branch OR on every release.
2. SBOMs in SPDX and CycloneDX formats are generated and stored in the repository.
3. SBOM files are attached to the release artifacts and indicated hyperlink in the changelog site.
4. The process is documented and easy to understand. Change log should have a hyperlink to the latest and greatest SBOM.

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_
```[tasklist]
### Tasks
- [ ] Create GitHub Actions Workflow
- [ ] Integrate SBOM Generation Tool
- [ ] Store and Attach SBOM
- [ ] Testing and Validation
- [ ] Documentation
```

### PR DESCRIPTION
**Pull request by rsh1k**
This pull request introduces a **Software Bill of Materials (SBOM)** test into our GitHub Actions workflow.

**Summary of Changes:**

1. Added a new file in the workflow to run the SBOM test.

**Details:**

1. The SBOM test step ensures that all dependencies used in the project are listed and validated against our security and compliance requirements.
2. This integration uses anchor-syft to generate and check the SBOM for any issues.

**Benefits:**

1. Enhanced Security: Helps identify and manage vulnerabilities in third-party dependencies.
2. Compliance: Assists in meeting regulatory and organizational requirements for software transparency.
3. Automation: Streamlines the process of verifying dependency information as part of our CI/CD pipeline.

### CODE REVIEW

File: .github/workflows/sbom.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+on:
```
Comment (mbiuki) [inline .github/workflows/sbom.yml:1]: Can you add some comments at the top about what this workflow does?

File: .github/workflows/sbom.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  release:
+    types: [published]
```
Comment (mbiuki) [inline .github/workflows/sbom.yml:3]: published or tagged - I believe we would like to have SBOM published w/ every release.

File: .github/workflows/sbom.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  workflow_dispatch:
+    inputs:
+      dotcms_version:
```
Comment (mbiuki) [inline .github/workflows/sbom.yml:6]: Should not be for one version only - any version release would need to have SBOM

**Comment by spbolton** [comment]
The PR checks will fail on the Sonarqube step here as this is a PR from a fork and forks do not have access to the Sonarqube token.   For the PR to be able to validate all the way through it needs to be a direct branch of dotcms/core.

**Comment by jdcmsd** [comment]
Steve's right; I slammed into that issue a few months ago, myself. The good news is that if all the changes are localized to one file, it'll be simple to pivot around this.

1. Close this pull request
2. Make a local clone of `dotcms/core` itself, rather than a fork: `git clone https://github.com/dotCMS/core.git`
3. Copy `.github/workflows/sbom.yml` from the fork to the new repo
4. Create a new branch via `git checkout -b YOUR_BRANCH_NAME`
5. Stage the local changes: `git add .github/workflows/sbom.yml` or `git add .`
6. Commit: `git git commit -m "include notes here!"
7. Push: `git push -u origin YOUR_BRANCH_NAME`
8. Browse to the branch you just pushed on GitHub, and create a new pull request from it.

All together, this should only take a few minutes. At that point, you're free to delete the fork, if you choose.

**Comment by mbiuki** [comment]
Rashik to create a new PR in dotCMS core and push from there - not from fork, thanks.

**Pull request closed without merging**